arm_gdb/mpu.py

```diff
@@ -0,0 +1,272 @@
+    regions_hex = type_str.split(':')[1].strip()
+    regions_value = int(regions_hex, 16)
+    regions = (regions_value >> 8) & 0xff
+    print(f"MPU region number {regions:#d}")
+
+    return regions
+
+def get_rbar_reg():
+    return [
+        RegisterDef("MPU_RBAR", "MPU Region Base Address Register", 0xE000ED9C, 4, [
+            FieldBitfield("BASE", 5, 27, "Starting address of MPU region address"),
+            FieldBitfieldEnum("SH", 3, 2, [
+                (0b00, False, "Non-shareable", None),
+                (0b01, False, "Outer shareable", None),
+                (0b10, False, "Inner Shareable", None),
+                ],"Shareability for Normal memory"),
+            FieldBitfieldEnum("AP", 1, 2, [
+                (0b00, False, "read/write by privileged code only", None),
+                (0b01, False, "read/write by any privilege level", None),
+                (0b10, False, "Read only by privileged code only", None),
+                (0b11, False, "Read only by any privilege level", None),
+                ],"Access permissions"),
+            FieldBitfieldEnum("XN", 0, 1, [
+                (0b1, False, "Disallow program execution in this region", None),
+                (0b0, False, "Allow program execution in this region", None),
+                ],"eXecute Never attribute"),
+            ]),
+    ]
+
+def get_rlar_reg():
+    return [
+        RegisterDef("MPU_RLAR", "MPU Region Base Limit Register", 0xE000EDA0, 4, [
+            FieldBitfield("LIMIT", 5, 27, "Ending address (upper inclusive limit) of MPU region address"),
+            FieldBitfieldEnum("PXN", 4, 1, [
+                (0b1, False, "Execution from a privileged mode is not permitted", None),
+                (0b0, False, "Execution only permitted if read permitted", None),
+                ],"Privileged execute-never"),
+            FieldBitfield("AttrIndx", 1, 3, "Attribute Index. Select memory attributes from attribute sets in MPU_MAIR0 and MPU_MAIR1"),
+            FieldBitfieldEnum("EN", 0, 1, [
+                (0b0, False, "disable", None),
+                (0b1, False, "Region enable", None),
+            ],"Region enable"),
+        ]),
+    ]
+
+def get_mpu_region_start():
+    REG_RBAR = 0xE000ED9C
+    value_str = gdb.execute(f"x/1x {REG_RBAR}", to_string=True)
+    value_hex = value_str.split(':')[1].strip()
+    value = int(value_hex, 16)
+    start = value & 0xFFFFFFE0;
+    return start
+
+def get_mpu_region_limit():
+    REG_RLAR = 0xE000EDA0
+    value_str = gdb.execute(f"x/1x {REG_RLAR}", to_string=True)
+    value_hex = value_str.split(':')[1].strip()
+    value = int(value_hex, 16)
+    limit = value & 0xFFFFFFE0;
+    return limit
+
+def mpu_region_enable():
+    REG_RLAR = 0xE000EDA0
+    value_str = gdb.execute(f"x/1x {REG_RLAR}", to_string=True)
+    value_hex = value_str.split(':')[1].strip()
+    value = int(value_hex, 16)
+    enabled = bool(value & 0x01)
+    return enabled
```

Comment:
The name of this function indicates to me that `enable`, is the action of enabling something.

What the function does is to check if the mpu region is enabled, for what I can tell. Given that it _reads_ a register using `x/1x` command.

I would prefer this to be named `mpu_current_region_is_enabled()`:
 - to indicate it's the current region (if I understand correctly?)
 - That it is a getter, (`_is_something` indicates get state of true or false)
 
 Also, use `read_mem()` as other places

Reply:
done